### PR TITLE
Enhancement: OpenTelemetry integration #180

### PR DESCRIPTION
**Pull request by cofin**
This enhancement enabled Open Telemetry integration.  It is set up very similar to the Compression backend. 

To use, set `instrumentation_config=InstrumentationConfig( backend='opentelemetry', excluded_urls="/test_path"  ),` in the Starlite application config.  

To do:
- Additional test cases
- Pass pre-commit pylint checks.  Current there is an `import-error` getting thrown.
- Verify metrics can be collected from third party utility
- Documentation

### CODE REVIEW

File: starlite/middleware/instrumentation/opentelemetry.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            excluded_urls=excluded_urls,
+            server_request_hook=server_request_hook,
+            client_request_hook=client_request_hook,
+            client_response_hook=client_response_hook,
+            tracer_provider=tracer_provider,
+            default_span_details=_get_route_details,
+        )
+
+
+def _get_route_details(
```
Comment (peterschutt) [inline starlite/middleware/instrumentation/opentelemetry.py:49]: Any thoughts on a work around? Could we use `scope["state"]`?

**Comment by peterschutt** [comment]
What do you use for ingestion? I might have a crack at getting the template app feeding into something like https://signoz.io/

**Comment by cofin** [comment]
> 

Signoz is the one that I see often.  I think it's specifically for APM.  I'd also like to try prometheus.

**Comment by cofin** [comment]
I'm going to close this and re-work the implementation.  I'll open a new PR once it's ready.

**Comment by seladb** [comment]
@cofin is there anything fundamentally wrong in the implementation that is in the closed PR? It looks quite ok to me... 

Just wondering if you had a completely different idea on how to implement this

**Pull request closed without merging**